cmd/earthly/main.go

```diff
@@ -1123,20 +1116,23 @@ func (app *earthlyApp) getAddressesForTCP(context *cli.Context) (addresses, erro
 		return addresses{}, err
 	}
 
-	lrURL, err := parseAndvalidateURL(app.cfg.Global.LocalRegistryHost)
-	if err != nil {
-		return addresses{}, err
+	if buildkitd.IsLocal(app.buildkitHost) && app.cfg.Global.LocalRegistryHost != "" {
+		// Local registry only matters when local, and specified.
+		lrURL, err := parseAndvalidateURL(app.cfg.Global.LocalRegistryHost)
+		if err != nil {
+			return addresses{}, err
+		}
+		if bkURL.Hostname() != lrURL.Hostname() {
+			app.console.Warnf("Buildkit and Local Registry URLs are pointed at different hosts (%s vs. %s)", bkURL.Hostname(), lrURL.Hostname())
+		}
 	}
 
 	if bkURL.Hostname() != dbURL.Hostname() {
 		app.console.Warnf("Buildkit and Debugger URLs are pointed at different hosts (%s vs. %s)", bkURL.Hostname(), dbURL.Hostname())
 	}
-	if bkURL.Hostname() != lrURL.Hostname() {
-		app.console.Warnf("Buildkit and Local Registry URLs are pointed at different hosts (%s vs. %s)", bkURL.Hostname(), lrURL.Hostname())
-	}
 
 	if bkURL.Hostname() == dbURL.Hostname() && bkURL.Port() == dbURL.Port() {
-		return addresses{}, errors.New("Debugger and Buildkit ports are the same")
+		return addresses{}, fmt.Errorf("Debugger and Buildkit ports are the same: %w", urlValidationFailure)
```

Comment:
Arguable use of this sentinel error. It fit well enough and I didn't want to add a single-use sentinel just for one test..